File: ios/Classes/Models/BasicIdMessage.swift
import Foundation

class BasicIdMessage: OdidMessage {
    let idType: Int
    let uaType: Int
    let uasId: String
    
    init(idType: Int, uaType: Int, uasId: String) {
        self.idType = idType
        self.uaType = uaType
        self.uasId = uasId
    }
    
    func getType() -> OdidMessageType {
        return .BASIC_ID
    }
    
    func toJson() -> Dictionary<String, Any> {
        return [
            "idType": idType,
            "uaType": uaType,
            "uasId": uasId
        ]
    }
    
    static func fromData(_ data: Data) -> BasicIdMessage {
        let bytes = Array(data)
        let typeByte = bytes[0]
        let idType = Int((typeByte & 0xF0) >> 4)
        let uaType = Int(typeByte & 0x0F)
        let uasId = String(bytes: bytes[1...], encoding: .ascii)
        
        return BasicIdMessage(
            idType: idType, uaType: uaType, uasId: uasId ?? ""
        )
    }
    
}

File: android/src/main/kotlin/cz/dronetag/flutter_opendroneid/models/BasicIdMessage.kt
package cz.dronetag.flutter_opendroneid.models

import cz.dronetag.flutter_opendroneid.OdidMessageHandler
import java.nio.ByteBuffer

class BasicIdMessage(
    var idType: Int,
    var uaType: Int,
    var uasId: String,
    override val type: OdidMessage.Type = OdidMessage.Type.BASIC_ID
) : OdidMessage {
    override fun toJson(): MutableMap<String, Any> {
        return mutableMapOf("idType" to idType, "uaType" to uaType, "uasId" to uasId)
    }

    companion object {
        fun fromBuffer(byteBuffer: ByteBuffer): BasicIdMessage {
            val type: Int = byteBuffer.get().toInt()
            val idType = type and 0xF0 shr 4
            val uaType = type and 0x0F
            var uasId = ByteArray(OdidMessageHandler.MAX_ID_BYTE_SIZE)
            byteBuffer.get(uasId, 0, OdidMessageHandler.MAX_ID_BYTE_SIZE)


            return BasicIdMessage(
                idType, uaType, String(uasId).trim()
            )
        }
    }
}